### PR TITLE
configure-gmsa.md updates

### PR DESCRIPTION
**Pull request by marosset**
Updating the container runtimes that support GMSA workloads on Windows.

All of the current container runtimes (Docker EE / Mirantis Container Runtime (via cri-dockerd) and containerd) have this support today.

/sig windows

### CODE REVIEW

File: content/en/docs/tasks/configure-pod-container/configure-gmsa.md [hunk 1]
```diff
@@ -197,18 +199,19 @@ As Pod specs with GMSA fields populated (as described above) are applied in a cl
...
 
-## Containerd 
+## Containerd
 
-On Windows Server 2019, in order to use GMSA with containerd, you must be running OS Build 17763.1817 (or later) which can be installed using the patch [KB5000822](https://support.microsoft.com/en-us/topic/march-9-2021-kb5000822-os-build-17763-1817-2eb6197f-e3b1-4f42-ab51-84345e063564). 
+On Windows Server 2019, in order to use GMSA with containerd, you must be running OS Build 17763.1817 (or later) which can be installed using the patch [KB5000822](https://support.microsoft.com/en-us/topic/march-9-2021-kb5000822-os-build-17763-1817-2eb6197f-e3b1-4f42-ab51-84345e063564).
 
-There is also a known issue with containerd that occurs when trying to connect to SMB shares from Pods. Once you have configured GMSA, the pod will be unable to connect to the share using the hostname or FQDN, but connecting to the share using an IP address works as expected. 
+There is also a known issue with containerd that occurs when trying to connect to SMB shares from Pods. Once you have configured GMSA, the pod will be unable to connect to the share using the hostname or FQDN, but connecting to the share using an IP address works as expected.
```
Comment (jsturtevant) [inline content/en/docs/tasks/configure-pod-container/configure-gmsa.md:206]: This was resolved in the March 2021 patch fix https://github.com/microsoft/Windows-Containers/issues/44#issuecomment-802404294 so can be dropped from docs as that is the recommended min version

Comment (marosset) [reply content/en/docs/tasks/configure-pod-container/configure-gmsa.md]: Good catch.
I made some updates for this (and hopefully they make sense).

**Review by celestehorgan — APPROVED** [review]
Ref for other reviewers: https://kubernetes.slack.com/archives/C0BP8PW9G/p1642628900084800 

Partially closes https://github.com/kubernetes/website/issues/30771

/lgtm

**Pull request merged**